src/main/java/org/apache/maven/shared/incremental/IncrementalBuildHelper.java

```diff
@@ -359,6 +378,143 @@ public void afterRebuildExecution( IncrementalBuildHelperRequest incrementalBuil
         }
 
     }
+    /**
+     * Detect whether the dependencies has changed since the last build.
+     *
+     * @param incrementalBuildHelperRequest
+     * @return <code>true</code> if the set of dependencies got changed since the last build.
+     * @throws MojoExecutionException
+     */
+    public boolean dependencyTreeChanged( IncrementalBuildHelperRequest incrementalBuildHelperRequest )
+            throws MojoExecutionException
+    {
+        File mojoConfigBase = getMojoStatusDirectory();
+        File mojoConfigFile = new File ( mojoConfigBase, DEPENDENCY_INFO_FILENAME );
+
+        classpathElements.clear();
+        dependencyInfo.clear();
+
+        final List<String> classpathElementsNew = incrementalBuildHelperRequest.getClasspathElements();
+
+        for ( String  classPathElement : classpathElementsNew )
+        {
+            File artifactPath = new File ( classPathElement );
+            if ( artifactPath.isFile() )
+            {
+                dependencyInfo.put( artifactPath, artifactPath.lastModified() );
+                classpathElements.add( artifactPath );
+            }
+        }
+        if ( dependencyInfo == null )
+        {
+            return false;
+        }
+        if ( !mojoConfigFile.exists() )
+        {
+            //no file, assume rebuild
+            return true;
+        }
+
+        Map<File, Long> origDependencyInfo;
+        List<File> origClasspathElements;
+        try
+        {
+            ObjectInputStream ois = new ObjectInputStream ( new FileInputStream( mojoConfigFile ) );
```

Comment:
we are java7 now so better using try resources or at least ois.close in finally block